mne/utils/check.py

```diff
@@ -255,7 +255,7 @@ def _check_fname(
     *,
     check_bids_split=False,
     verbose=None,
-):
+) -> Path:
```

Comment:
I added this type-hint with the hope that we could start validating this one since `_check_fname` is our main validation function to check for file/directory existence and to cast to `Path`.